test/functional/specs/C14409.js

```diff
@@ -0,0 +1,96 @@
+import { t, ClientFunction } from "testcafe";
+import createNetworkLogger from "../helpers/networkLogger";
+import getResponseBody from "../helpers/networkLogger/getResponseBody";
+import fixtureFactory from "../helpers/fixtureFactory";
+import cookies from "../helpers/cookies";
+import alloyEvent from "../helpers/alloyEvent";
+
+const networkLogger = createNetworkLogger();
+
+fixtureFactory({
+  title: "C14409 - Consenting to no purposes should be persisted.",
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+
+test.meta({
+  ID: "C14409",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const setConsentToNone = ClientFunction(() => {
+  return window.alloy("setConsent", { general: "out" });
+});
+
+test("C14409 - Consenting to no purposes should be persisted.", async () => {
+  const imsOrgId = "53A16ACB5CC1D3760A495C99@AdobeOrg";
+  await cookies.clear();
+
+  const configure = await alloyEvent("configure", {
+    configId: "9999999",
+    orgId: imsOrgId,
+    debugEnabled: true,
+    idMigrationEnabled: false
+  });
+
+  await configure.promise;
+
+  // send alloy event
+  const event1 = await alloyEvent({
+    viewStart: true
+  });
+
+  // apply user consent
+  await setConsentToNone();
+
+  await event1.promise;
+
+  const cookieName = `kndctr_${imsOrgId.replace(
+    /[@]+?/,
+    "_"
+  )}_consent`.toString();
+
+  const consentCheck = await cookies.get(cookieName);
+
+  await t.expect(consentCheck).eql("general=out");
+
+  // test konductor response
+  await t.expect(networkLogger.edgeEndpointLogs.requests.length).eql(1);
+  await t
+    .expect(networkLogger.edgeEndpointLogs.requests[0].response.statusCode)
+    .eql(200);
+
+  const body = getResponseBody(networkLogger.edgeEndpointLogs.requests[0]);
+  const bodyObject = JSON.parse(body);
+
+  // read state:store handles from response (i.e. 'set a cookie')
+  const cookiesToSet = {};
+
+  await t.expect("handle" in bodyObject).ok();
+  await t.expect(bodyObject.handle.length).gt(0);
+
+  await bodyObject.handle.map(handle => {
+    if (handle.type !== "state:store") return handle;
+    handle.payload.map(payload => {
+      cookiesToSet[payload.key] = payload;
+      return payload;
+    });
+    return handle;
+  });
```

Comment:
I think we should be using `createResponse(response).getPayloadsByType("store:handle");` then do a `reduce` to build the `cookiesToSet` object:

```
const storePayloads = createResponse(response).getPayloadsByType("store:handle");
const cookiesToSet = storePayloads.reduce((memo, storePayload) => {
  memo[storePayload.key] = storePayload;
  return memo;
, {});
```

It's possible to use a `forEach` instead of a `reduce` for building the `cookiesToSet` object like so:

```
const cookiesToSet = {};
storePayloads.forEach(storePayload => {
  cookiesToSet[storePayload.key] = storePayload;
}
```

but we had a team discussion about which pattern we wanted to follow for consistency and `reduce` received the majority support.

You'll probably want to update C14407.js with whatever you end up doing here.